main/main.go

```diff
@@ -21,117 +21,237 @@ import (
-var localTimeout = flag.Duration("localTimeout", 0, "Timeout to set locally (read+write)")
-var localReadTimeout = flag.Duration("localReadTimeout", 0, "Timeout to set locally (read)")
-var localWriteTimeout = flag.Duration("localWriteTimeout", 0, "Timeout to set locally (write)")
-var remoteTimeout = flag.Duration("remoteTimeout", 0, "Timeout to set for remote redises (connect+read+write)")
-var remoteReadTimeout = flag.Duration("remoteReadTimeout", 0, "Timeout to set for remote redises (read)")
-var remoteWriteTimeout = flag.Duration("remoteWriteTimeout", 0, "Timeout to set for remote redises (write)")
+const DEFAULT_POOL_SIZE = 20
+
+var host                 = flag.String("host", "localhost", "The host to listen for incoming connections on")
+var port                 = flag.Int("port", 6379, "The port to listen for incoming connections on")
+var socket               = flag.String("socket", "", "The socket to listen for incoming connections on.  If this is provided, host and port are ignored")
+var maxProcesses         = flag.Int("maxProcesses", 0, "The number of processes to use.  If this is not defined, go's default is used.")
+var poolSize             = flag.Int("poolSize", DEFAULT_POOL_SIZE, "The size of the connection pools to use")
+var tcpConnections       = flag.String("tcpConnections", "localhost:6380 localhost:6381", "TCP connections (destination redis servers) to multiplex over")
+var unixConnections      = flag.String("unixConnections", "", "Unix connections (destination redis servers) to multiplex over")
+var localTimeout         = flag.Duration("localTimeout", 0, "Timeout to set locally (read+write)")
+var localReadTimeout     = flag.Duration("localReadTimeout", 0, "Timeout to set locally (read)")
+var localWriteTimeout    = flag.Duration("localWriteTimeout", 0, "Timeout to set locally (write)")
+var remoteTimeout        = flag.Duration("remoteTimeout", 0, "Timeout to set for remote redises (connect+read+write)")
+var remoteReadTimeout    = flag.Duration("remoteReadTimeout", 0, "Timeout to set for remote redises (read)")
+var remoteWriteTimeout   = flag.Duration("remoteWriteTimeout", 0, "Timeout to set for remote redises (write)")
 var remoteConnectTimeout = flag.Duration("remoteConnectTimeout", 0, "Timeout to set for remote redises (connect)")
-var cpuProfile = flag.String("cpuProfile", "", "Direct CPU Profile to target file")
+var cpuProfile           = flag.String("cpuProfile", "", "Direct CPU Profile to target file")
+var configFile           = flag.String("config", "", "Configuration file (JSON)")
 
 func main() {
 	flag.Parse()
 
+	var configs []PoolConfig
+	var err error
+
 	if *cpuProfile != "" {
 		f, err := os.Create(*cpuProfile)
-		if err == nil {
-			pprof.StartCPUProfile(f)
-			defer pprof.StopCPUProfile()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error when creating cpu profile file: %s\r\n", err)
+			os.Exit(1)
```

Comment:
move terminate code to new function like terminateIfError(err) and call this function wherever needed.